examples/physics/basic/example.js

```diff
@@ -0,0 +1,265 @@
+    let d = dist(position.x, position.y, position.z, this.position.x, this.position.y, this.position.z);
+    return (d <= radius + this.radius)
+  }
+
+
+  // if the ball has intersected with a boundary, check the dot() product after a collision
+  // a dot product is a way of asking whether two vectors are facing the same direction
+  // in this case, if it's less 0, then no. otherwise, yes. 
+  checkBoundary(boundary) {
+    if (boundary.checkIntersection(this.position, this.radius)) {
+      if (p5.Vector.dot(boundary.reflectionNormal, this.velocity) <= 0) {
+        return;
+      }
+      // velocity of the ball is reversed upon collision
+      // this causes the ball to bounce back up
+      this.velocity.mult(boundary.reflectionNormal);
+
+      // change the color of the sphere everytime it hits the ground
+      this.color = color(random(255), random(255), random(255));
+
+    }
+  }
+
+  render() {
+    push();
+    // if the ball is held...
+    // make it a different color than normal
+    if (this.held) {
+      fill(this.heldColor);
+    } else {
+      fill(this.color);
+    }
+    noStroke();
+
+    // translates the ball to it's position
+    translate(this.position);
+
+    // adding this material makes it easier to see edges of the boundaries
+    emissiveMaterial(this.color)
+
+    // size of sphere
+    sphere(this.radius);
+    pop();
+  }
+}
+
+class Boundary {
+  // create a boundary class for the walls/ground
+  constructor(
+    x,
+    y,
+    z,
+    width,
+    height,
+    depth,
+    reflectionNormalX,
+    reflectionNormalY,
+    reflectionNormalZ
+  ) {
+
+    // gives boundaries their own x, y, z position
+    this.position = createVector(x, y, z);
+
+    // gives a minimum and maximum height, width, and depth to the boundaries
+    this.scale = createVector(width, height, depth);
+
+    // create a force to reflect any incoming balls
+    this.reflectionNormal = createVector(
+      reflectionNormalX,
+      reflectionNormalY,
+      reflectionNormalZ
+    );
+
+    // designated minimum and maximum width, height, and depth of the boundaries
+    this.minY = y - height / 2;
+    this.maxY = y + height / 2;
+    this.minX = x - width / 2;
+    this.maxX = x + width / 2;
+    this.minZ = z - depth / 2;
+    this.maxZ = z + depth / 2;
+    this.color = color(50, 200, 50);
+    this.friction = 0.98;
+  }
+
+
+  // checks intersection of the ball with x, y, z positions of the boundaries
+  checkIntersection(position, radius) {
+    const x = Math.max(this.minX, Math.min(position.x, this.maxX));
+    const y = Math.max(this.minY, Math.min(position.y, this.maxY));
+    const z = Math.max(this.minZ, Math.min(position.z, this.maxZ));
+
+    var distance = Math.sqrt(
+      (x - position.x) * (x - position.x) +
+      (y - position.y) * (y - position.y) +
+      (z - position.z) * (z - position.z)
+    );
+    return distance < radius;
+  }
+
+  render() {
+    push();
+    if (this.held) {
```

Comment:
I don't think that a boundary is ever `held`. you can just change this to `fill(this.color)` and scrap the conditional.